pennylane/gradients/parameter_shift_cv.py

```diff
@@ -0,0 +1,690 @@
+        return op.grad_method
+
+    if op.grad_method != "A":
+        raise ValueError(f"Operation {op} has unknown gradient method {op.grad_method}")
+
+    # Operation supports the CV parameter-shift rule.
+    # Create an empty list to store the 'best' partial derivative method
+    # for each observable
+    best = []
+
+    for m in tape.measurements:
+
+        if (m.return_type is qml.operation.Probability) or (m.obs.ev_order not in (1, 2)):
+            # Higher-order observables (including probability) only support finite differences.
+            best.append("F")
+            continue
+
+        # get the set of operations betweens the operation and the observable
+        ops_between = tape.graph.nodes_between(op, m.obs)
+
+        if not ops_between:
+            # if there is no path between the operation and the observable,
+            # the operator has a zero gradient.
+            best.append("0")
+            continue
+
+        # For parameter-shift compatible CV gates, we need to check both the
+        # intervening gates, and the type of the observable.
+        best_method = "A"
+
+        if any(not k.supports_heisenberg for k in ops_between):
+            # non-Gaussian operators present in-between the operation
+            # and the observable. Must fallback to numeric differentiation.
+            best_method = "F"
+
+        elif m.obs.ev_order == 2:
+
+            if m.return_type is qml.operation.Expectation:
+                # If the observable is second order, we must use the second order
+                # CV parameter shift rule
+                best_method = "A2"
+
+            elif m.return_type is qml.operation.Variance:
+                # we only support analytic variance gradients for
+                # first order observables
+                best_method = "F"
+
+        best.append(best_method)
+
+    if all(k == "0" for k in best):
+        # if the operation is independent of *all* observables
+        # in the circuit, the gradient will be 0
+        return "0"
+
+    if "F" in best:
+        # one non-analytic observable path makes the whole operation
+        # gradient method fallback to finite-difference
+        return "F"
+
+    if "A2" in best:
+        # one second order observable makes the whole operation gradient
+        # require the second order parameter-shift rule
+        return "A2"
+
+    return "A"
+
+
+def _gradient_analysis(tape):
+    """Update the parameter information dictionary of the tape with
+    gradient information of each parameter."""
+
+    if getattr(tape, "_gradient_fn", None) is param_shift_cv:
+        # gradient analysis has already been performed on this tape
+        return
+
+    tape._gradient_fn = param_shift_cv
+
+    for idx, info in tape._par_info.items():
+        info["grad_method"] = _grad_method(tape, idx)
+
+
+def _transform_observable(obs, Z, device_wires):
+    """Apply a Gaussian linear transformation to each index of an observable.

```

Comment:
The transformation is applied to the observable itself, right, not the index?

Reply:
Yes I think you're correct
```suggestion
    """Apply a Gaussian linear transformation to an observable.
```